Visual IaC Import Blog Post (#14967)
pulumi/docs: content/blog/visual-iac-import/index.md

---
title: "Go from Cloud to Code in Minutes with Visual IaC Import"
date: 2025-05-08T14:41:53-05:00
draft: false
meta_desc: Introducing Pulumi Visual IaC Import—discover, codify, and generate clean infrastructure code with AI assistance.

> meta_image: visual-iac-import.png

Reviewer: I think we'll need the title in the meta image.

authors:
    - insights-team
tags:
    - iac
    - import
    - insights

social:
    twitter:
    linkedin:

---

Pulumi is excited to announce the release of **Visual IaC Import**, a new feature that transforms the way platform teams onboard existing cloud infrastructure into Pulumi. With Visual IaC Import, you can discover unmanaged cloud resources across your organization, organize them into logical units, and instantly generate high-quality Pulumi code—with AI enhancements built in.

> Visual IaC Import is now available for all Team and Enterprise customers with Pulumi Insights enabled.

Reviewer 2: We should include Business Critical customers. So 'for all Team, Enterprise and Business Critical' customers...

One of the biggest challenges platform teams face when adopting infrastructure as code is migrating existing cloud deployments. Legacy scripts, manual configurations, and scattered resource ownership all slow down progress. Visual IaC Import solves this by making the journey from cloud to code seamless and intuitive.

Instead of manually digging through consoles and writing boilerplate code by hand, Visual IaC Import enables teams to:

* **Discover** all cloud resources—across any number of accounts—via Pulumi Insights.
* **Select** and group resources into meaningful stacks, services, or projects.
* **Generate** production-ready Pulumi code in your language of choice.
* **Enhance** that code using Pulumi Copilot, which refines formatting, naming, references, and documentation.
* **Integrate** with your development workflows by downloading, editing, and importing Pulumi code into your existing repositories.

Whether you’re modernizing infrastructure, unifying multi-account environments, or enabling developers to adopt infrastructure as code more easily, Visual IaC Import gives you a fast, repeatable onramp to scalable, consistent cloud management.

## How It Works

> Visual IaC Import is a guided three-step workflow accessible via the Pulumi Cloud Console:

Reviewer: ```suggestion
Visual IaC Import is a guided three-step workflow within the Pulumi Cloud Console:
```

1. Select Resources: Use powerful filters and grouping tools to find the resources you want to codify. Whether you're capturing a full service or just a few related components, Visual IaC Import gives you visibility and precision.
2. Review References: Automatically discover and select dependent resources—ensuring your code captures the right context and connectivity between infrastructure elements.
3. Generate Code: Choose your preferred language and preview or download fully-formed Pulumi code. You can refine it manually or invoke Pulumi Copilot to improve naming, structure, and comments.

![Visual IaC Code Generation](visual-iac-import.png)

> Throughout the workflow, you maintain full control. Select just what you need, tweak it as required, and integrate on your terms.  You can find full documentation and usage details in the [Visual IaC Import docs](https://www.pulumi.com/docs/pulumi-cloud/import/).

Reviewer: ```suggestion
Throughout the workflow, you maintain full control. Select what you need, tweak it as required, and integrate on your terms.  You can find full documentation and usage details in the [Visual IaC Import docs](/docs/pulumi-cloud/import/).
```

Once resources are in code, you can take full advantage of Pulumi Cloud’s full suite of features: define and enforce policies with CrossGuard, gain visibility into your entire infrastructure through Pulumi Insights, manage state with ESC’s secrets management, and automate workflows through Pulumi Deployments.

By codifying your infrastructure, you reduce the risk of configuration drift, simplify onboarding, and ensure reproducibility across staging and production.

## From Insights to IaC—Seamlessly

Visual IaC Import builds on Pulumi Insights, providing a natural path from discovery to codification. If you're already using Insights to monitor and visualize your infrastructure, you can now convert those resources into managed, versioned code in just a few clicks.

And with built-in support for dependency analysis and cross-account visibility, you can finally treat your entire infrastructure portfolio as a single, coherent system—even if it wasn't built that way.

## Try It Today

> Visual IaC Import is now available in the Pulumi Cloud Console for Team and Enterprise users. If you’re ready to modernize your infrastructure and empower your teams with a fast path to Infrastructure as Code, there’s no better place to start.

Reviewer 2: List above, we need to add Business Critical to the list of users, "Pulumi Cloud Console for Team, Enterprise and Business Critical users."
